tests/testenv/mod.rs

```diff
@@ -162,6 +162,25 @@ impl TestEnv {
         }
     }
 
+    /// Create a broken symlink at the given path in the temp_dir.
+    pub fn create_broken_symlink<P: AsRef<Path>>(
+        &mut self,
+        link_path: P,
+    ) -> Result<PathBuf, io::Error> {
+        let root = self.test_root();
+        let broken_symlink_link = root.join(link_path);
+        {
+            let temp_target_dir = TempDir::new("fd-tests-broken-symlink")?;
```

Comment:
Do we really have to create this directory? I would assume that we can just create a symlink to a non-existing directory by passing a non-existing path?